modules/missionmail.js

```diff
@@ -0,0 +1,258 @@
+          } else {
+            parsed.text = parsed.subject;
+          }
+        }
+        if (!parsed.text) {throw new Error("unable to parse email with no discrnable text, html, or subject");}
+        // trim the reply quote from the bottom if there is one (for some reason it was bypassing email replace)
+        for (const regex of replyRegexes) {
+          const match = parsed.text.toLowerCase().search(regex);
+          if (match && match > 0) {
+            // parsed.fullText = parsed.text;
+            parsed.text = parsed.text.substring(0, match);
+            break; // Stop after the first match to avoid over-trimming
+          }
+        }
+        parsed.text = parsed.text.trimEnd();
+        // search for profanity
+        const pfViolations = pf.scan(parsed.text + "");
+        const bannedViolations = [banned.links, banned.words, banned.scam].flat().filter(bannedString => parsed.text.includes(bannedString) ? bannedString : undefined);
+        // figure out who it is from
+        const fromEmail = parsed.from ? parsed.from[0].address : "Err:NoFromAddress";
+        const missionaryId = await u.db.sheets.missionaries.findKey(address => fromEmail?.includes(address) ? address : false);
+        // get some discord side of things stuff
+        const ldsg = await module.exports.client.guilds.fetch(u.sf.ldsg);
+        const missionary = missionaryId ? await ldsg.members.fetch(missionaryId) : undefined;
+        const missionMailApprovals = await ldsg.channels.fetch(u.sf.channels.missionMailApprovals);
+        if (!missionMailApprovals || missionMailApprovals.type !== ChannelType.GuildText) { throw new Error("unable to find approval channel for missionary emails."); }
+        // setup the functions to mark it as handled and forward it when approved, or just mark when rejected
+        // setup the request message, including listing detected profanity at the top,
+        // and the embed almost the same, but not fully replaced, just marked as will be replaced.
+        // and approve and reject buttons
+        const approveBtn = new u.Button().setCustomId(approveIdPrefix + "from" + rawMsg.uid).setLabel("Approve").setStyle(ButtonStyle.Primary);
+        const rejectBtn = new u.Button().setCustomId(rejectIdPrefix + "from" + rawMsg.uid).setLabel("Reject").setStyle(ButtonStyle.Danger);
+        const actionRow = u.MessageActionRow().addComponents([approveBtn, rejectBtn]);
+        const requestMsg = {
+          components: [actionRow],
+          content:
+            (bannedViolations.length > 0 ? '# DETECTED BANNED PHRASES:\n' + bannedViolations.join(', ') : '')
+            + (bannedViolations.length > 0 && pfViolations.length > 0 ? '\n' : '') +
+            (pfViolations.length > 0 ? '# DETECTED PROFANITY:\n' + pfViolations.join(', ') : ''),
+          embeds: [u.embed({
+            title: `incoming missionary email from ${missionary?.user.username ?? "NON REGISTERED MISSIONARY EMAIL"}(${fromEmail}) - ${parsed.subject}`,
```

Comment:
If icarus can't find the missionary, it probably shouldn't be sending it in the first place